tangos/live_calculation/builtin_functions/link.py

```diff
@@ -20,14 +20,18 @@ def link(*args):
     raise RuntimeError("Internal error in link(). The customised function from the link_initialisation routine should be called.")
 
 link.set_input_options(0, provide_proxy=True, assert_class=StoredProperty)
-link.set_input_options(1, provide_proxy=True, assert_class=StoredProperty)
+link.set_input_options(1, provide_proxy=True)
```

Comment:
I can't remember exactly why the class of 1 had to be `StoredProperty` but equally is there a clear reason for removing that requirement?

Also I am concerned that since the other N arguments don't have `provide_proxy=True` the code may end up trying to fetch the values of say `BH_mass` before actually calling the `link` function. Probably won't break anything but could be wasteful.

In other words we might want something like
```python
link.set_input_options(2, assert_class=FixedInput)
for i in range(20): # TODO: implement a way to set provide_proxy on all possible arguments, to remove arbitrary upper limit
    link.set_input_options(i, provide_proxy=True)
```

It's ugly but should make things more efficient

Reply:
Ok. I agree... a bit ugly, but I see your point

Reply:
Third input is the string 'max' or 'min' no?

Yes I agree about the first input and also it should presumably be `range(3,20)` rather than `range(20)` which would overwrite the options for 0,1,2